rosetta-source/src/main/java/cdm/base/staticdata/codelist/LoadCodeListImpl.java

```diff
@@ -0,0 +1,104 @@
+package cdm.base.staticdata.codelist;
+
+import cdm.base.staticdata.codelist.functions.LoadCodeList;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.regnosys.rosetta.common.serialisation.RosettaObjectMapper;
+
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.URISyntaxException;
+import java.net.URL;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.Set;
+import java.util.jar.JarEntry;
+import java.util.jar.JarFile;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+public class LoadCodeListImpl extends LoadCodeList {
+
+    /** Directory path where JSON CodeLists are stored inside the resources. */
+    private static final String CODELIST_COLLECTION_PATH = "org/isda/codelist/json";
+    private static final ObjectMapper mapper = RosettaObjectMapper.getNewRosettaObjectMapper();
```

Comment:
Potential memory leak. @hugohills-regnosys not entirely sure about this one. What do you think?